neo/UIntBase.cs

```diff
@@ -61,7 +62,12 @@ public bool Equals(UIntBase other)
                 return true;
             if (data_bytes.Length != other.data_bytes.Length)
                 return false;
-            return data_bytes.SequenceEqual(other.data_bytes);
+
+            for (int x = data_bytes.Length - 1; x >= 0; x--)
+            {
+                if (data_bytes[x] != other.data_bytes[x]) return false;
```

Comment:
good.

Reply:
Do they differ a lot?

Reply:
I am not sure, @erikzhang.
But @igormcoelho, @shargon and @jsolman discussed that some that ago.

The description of "SequenceEqual" says: "Determines whether two sequences are equal by comparing the elements by using the default equality comparer for their type."

The default comparer should be `==`: 
https://github.com/neo-project/neo/blob/9b71d5f63de901ca83b4005c40f1ca9099ff39f8/neo/UIntBase.cs#L176-L183
Which calls: `Equals`: https://github.com/neo-project/neo/blob/9b71d5f63de901ca83b4005c40f1ca9099ff39f8/neo/UIntBase.cs#L77-L84

By the way, I just saw that `==` already checks `ReferenceEquals` (which was redudant in the previous version).

I believe it is better like this, @igormcoelho will have a better vision.

Reply:
> Determines whether two sequences are equal by comparing the elements by using the default equality comparer for their type.

Did you see the work **"elements"**? That means it calls the default equality comparer of the **"elements"**.

Reply:
That is true, @erikzhang, you are right, good interpretation :dagger:, master!
In this sense, it would call a comparer of `byte` per `byte`, which would be the same, yes?

Reply:
Yes. I think so.

Reply:
So, let's revert it.

If you want to check it further let us know, @shargon.

Reply:
@shargon, for reference, let's also consider these questions and response on linq issue.

https://github.com/neo-project/neo/issues/939#issuecomment-513197781

https://github.com/neo-project/neo/issues/939#issuecomment-528173250